### PR TITLE
Stage 3

### PR DESCRIPTION
**Pull request by rawisa**

### CODE REVIEW

File: firefox-ios/README.md [hunk 1]
```diff
@@ -167,4 +167,46 @@ general detailed code structure can be found at (note: this code structure are f
   Following [Swift style]( https://github.com/raywenderlich/swift-style-guide) which was recommended by firefox. We added the new feature, and created a pull request and attached it on there bugzilla site according to there [guidelines](https://github.com/mozilla/firefox-ios#contributor-guidelines)
   
   * [Forked repo.](https://github.com/rawisa/firefox-ios)
-  * [PR](https://github.com/mozilla/firefox-ios/pull/1889) For the missing feature.
\ No newline at end of file
+  * [PR](https://github.com/mozilla/firefox-ios/pull/1889) For the missing feature.
```
Comment (robi-y) [inline firefox-ios/README.md:171]: very nice 👍

File: firefox-ios/README.md [hunk 2]
```diff
@@ -167,4 +167,46 @@ general detailed code structure can be found at (note: this code structure are f
...
+  
+  
+## Metrics, Variability and Quality Measures
```
Comment (robi-y) [inline firefox-ios/README.md:174]: see also, details at: http://aosabook.org/en/ffreleng.html

File: firefox-ios/README.md [hunk 3]
```diff
@@ -167,4 +167,46 @@ general detailed code structure can be found at (note: this code structure are f
...
+  
+  It ran 263 tests in xcode with 0 failures. When the tests finishes it shows a green arrow on the PR to verify that our build passed all tests:
+  
+  ![](our-pr.png)
+  
+  
+## Security
+  ![](security.jpg)
+  
+  **Firefox 4:**
```
Comment (robi-y) [inline firefox-ios/README.md:255]: Is it specific to firefox-ios?

**Comment by robi-y** [comment]
Nice work, see some inline comments.
I miss some discussion about metrics, can you find information about applying metric tools to find modeling or other quality properties of the code.

**Comment by rawisa** [comment]
Finished, feel free to merge if everything is good now ;)

File: firefox-ios/README.md [hunk 4]
```diff
@@ -45,6 +45,7 @@ They can find the bugs that need to be fixed at the following list:
 
 * Bugs:           [File a new bug](https://bugzilla.mozilla.org/enter_bug.cgi?bug_file_loc=http%3A%2F%2F&bug_ignored=0&op_sys=iOS%20&product=Firefox%20for%20iOS&rep_platform=All) • [Existing bugs](https://bugzilla.mozilla.org/describecomponents.cgi?product=Firefox%20for%20iOS) 
 
+* Another ASOSMA project: http://aosabook.org/en/ffreleng.html
```
Comment (robi-y) [inline firefox-ios/README.md:48]: This other project is called AOSA and it is actually the inspiration for ours

**Comment by robi-y** [comment]
Nice, how hard is it to add new features or modules to ff-ios? can you elaborate a little about it

**Pull request merged**